exercise/static/exercise/user_results.js

```diff
@@ -0,0 +1,102 @@
+function startListen() {
+	if ($(".module-to-collapse").length > 0) {
+		// Show the "Expand all modules" button if some modules are collapsed
+		$("#toggle-expand-all-modules").show();
+		// Check local storage and expand all modules if the variable "expandAllModules" is set to true
+		const expandAllModules = localStorage.getItem("expandAllModules") === "true";
+		if (expandAllModules) {
+			$(".module-panel > div").addClass("in").collapse("show"); // addClass("in") skips the expand animation
+			$("#toggle-expand-all-modules").find("span").removeClass("glyphicon-unchecked").addClass("glyphicon-check");
+		}
+
+		// Register an event handler to the "Expand all modules" button
+		$("#toggle-expand-all-modules").on("click", function(event) {
+			const button = $(this);
+			const expandAllModules = localStorage.getItem("expandAllModules") === "true";
+			if (expandAllModules) {
+				$(".module-panel, .module-to-collapse > div").collapse("hide");
+				button.find("span").removeClass("glyphicon-check").addClass("glyphicon-unchecked");
+				localStorage.setItem("expandAllModules", false);
+			} else {
+				$(".module-panel > div").collapse("show");
+				button.find("span").removeClass("glyphicon-unchecked").addClass("glyphicon-check");
+				localStorage.setItem("expandAllModules", true);
+			}
+		});
+	}
+
+	// Check automatic scrolling behavior and show appropriate button
+	const autoScrollBehavior = localStorage.getItem("autoScrollBehavior");
+	if (autoScrollBehavior === "instant") {
+		$("#auto-scroll-behavior-instant").show();
+	} else if (autoScrollBehavior === "off") {
+		$("#auto-scroll-behavior-off").show();
+	} else {
+		$("#auto-scroll-behavior-smooth").show();
+		localStorage.setItem("autoScrollBehavior", "smooth"); // Scroll smoothly by default
+	}
+
+	// Register event handlers to all three scroll behavior buttons
+	$("#cycle-auto-scroll-behavior button").on("click", function(event) {
+		$(this).hide();
+		const autoScrollBehavior = localStorage.getItem("autoScrollBehavior");
+		if (autoScrollBehavior === "instant") {
+			$("#auto-scroll-behavior-off").show();
+			localStorage.setItem("autoScrollBehavior", "off");
+		} else if (autoScrollBehavior === "off") {
+			$("#auto-scroll-behavior-smooth").show();
+			localStorage.setItem("autoScrollBehavior", "smooth");
+		} else {
+			$("#auto-scroll-behavior-instant").show();
+			localStorage.setItem("autoScrollBehavior", "instant");
+		}
+	});
```

Comment:
This has the same duplication issue as the expand all button.

Reply:
Fixed by adding a separate button for enabling/disabling auto-scrolling and setting the scroll behavior.